plugins/telemetry/telemetry_impl.cpp

```diff
@@ -489,6 +509,39 @@ void TelemetryImpl::process_heartbeat(const mavlink_message_t &message)
     }
 }
 
+void TelemetryImpl::process_statustext(const mavlink_message_t &message)
+{
+    mavlink_statustext_t statustext;
+    mavlink_msg_statustext_decode(&message, &statustext);
+
+    Telemetry::StatusText::StatusType type;
+
+    switch (statustext.severity) {
+        case MAV_SEVERITY_WARNING:
+            type = Telemetry::StatusText::StatusType::WARNING;
+            break;
+        case MAV_SEVERITY_CRITICAL:
+            type = Telemetry::StatusText::StatusType::CRITICAL;
+            break;
+        default:
+            type = Telemetry::StatusText::StatusType::INFO;
+            break;
+    }
+
+    // statustext.text is not null terminated, therefore we copy it first to
```

Comment:
Are we sure about that? Looks like something that could be inspired from somewhere else, so just wanting to confirm for `statustext.text` :)

Reply:
Unfortunately that's true:
https://mavlink.io/en/messages/common.html#STATUSTEXT

And even if it was not true, it's a good precaution anyway.